relays/bin-substrate/src/cli/encode_message.rs

```diff
@@ -56,41 +85,45 @@ pub trait CliEncodeMessage: Chain {
 }
 
 /// Encode message payload passed through CLI flags.
-pub(crate) fn encode_message<Source: Chain, Target: Chain>(
+pub(crate) fn encode_message<Source: CliEncodeMessage, Target: Chain>(
 	message: &Message,
 ) -> anyhow::Result<RawMessage> {
 	Ok(match message {
 		Message::Raw { ref data } => data.0.clone(),
 		Message::Sized { ref size } => {
-			let expected_xcm_size = match *size {
+			let destination = match Target::ID {
```

Comment:
can we use `try_from_chain_id` here like:
```
match CustomNetworkId::try_from_chain_id(Target::ID) {
    Some(network_id) => network_id,
    _ => return Err(anyhow::format_err!("Unsupported target chain: {:?}", Target::ID)),
}

```

Reply:
and/or maybe implement `TryFrom for CustomNetworkId`

Reply:
> can we use try_from_chain_id here? Something like:

Indeed, I've missed that. Thanks

> and/or maybe implement TryFrom for CustomNetworkId

The idea was to make the pairing fn for `as_network_id`, which is `const fn`, but right now we don't need it. So changed to the idiomatic way